src/Ml.Cli.WebApp/Server/Datasets/Database/FileStorage/FileService.cs

```diff
@@ -110,8 +174,32 @@ private async Task<BlobClient> CloudBlockBlob(string blobStorageName, string con
         return blockBlob;
     }
     
-    public async Task<IList<string>> GetImportedDatasetsNamesAsync(string blobStorageName, string containerName)
+    private static async Task<IList<string>> ListBlobsHierarchicalListing(BlobContainerClient container, 
+        string prefix, 
+        int? segmentSize, int maxLevel =2, int level=0)
     {
+        var resultSegment = container.GetBlobsByHierarchyAsync(prefix:prefix, delimiter:"/")
+                .AsPages(default, segmentSize);
+            var directories = new List<string>();
+            await foreach (var blobPage in resultSegment)
+            {
+                foreach (var blobhierarchyItem in blobPage.Values)
+                {
+                    if (!blobhierarchyItem.IsPrefix) continue;
+                    directories.Add(blobhierarchyItem.Prefix.Substring(0, blobhierarchyItem.Prefix.Length-1));
+                    if (level >= maxLevel) continue;
+                    var result =await ListBlobsHierarchicalListing(container, blobhierarchyItem.Prefix, null, maxLevel, level+1);
+                    directories.AddRange(result);
+                }
+            }
+
+            return directories;
+    }
+    
+    public async Task<IList<string>> GetImportedDatasetsNamesAsync(string blobDirectoryUri)
```

Comment:
"blobDirectoryUri" parameter is not the same as "blobUri", which is defined in the interface